doc/swish/parallel.tex

```diff
@@ -0,0 +1,183 @@
+\label{parallel:options}
+
+\begin{tabular}{lp{5em}p{.65\textwidth}}
+  option & default & description \\ \hline
+
+  \code{start-limit}
+  & \code{(most-positive-fixnum)}
+  & a positive fixnum; the number of workers allowed to start and run
+  concurrently \\
+
+  \code{order}
+  & \code{random}
+  & a symbol, one of \code{random}, \code{left}, or \code{right} \\
+
+  \code{timeout}
+  & \code{infinity}
+  & \code{infinity} or a nonnegative fixnum; the number of
+  milliseconds the workers are allowed to execute before failing the
+  entire operation \\
+\end{tabular}
+
+\defineentry{parallel}
+\begin{syntax}
+  \code{(parallel () \var{e} \etc)}\\
+  \code{(parallel ([\var{option} \var{value}] \etc) \var{e} \etc)}\\
+  \code{(parallel ,\var{options} \var{e} \etc)}
+\end{syntax}
+
+The \code{parallel} construct invokes each expression \var{e}, \etc{}
+concurrently and returns a list of the resulting values. Expressions
+may run concurrently unless \code{start-limit} is 1.
+
+The \code{(parallel ([\var{option} \var{value}] \etc) \var{e} \etc)}
+form constructs a copy of the default options overriding the specified
+\var{option} with the specified \var{value}.
+
+\defineentry{parallel"!}
+\begin{syntax}
+  \code{(parallel! () \var{e} \etc)}\\
+  \code{(parallel! ([\var{option} \var{value}] \etc) \var{e} \etc)}\\
+  \code{(parallel! ,\var{options} \var{e} \etc)}
+\end{syntax}
+
+The \code{parallel!} construct behaves as \code{parallel} but returns
+an unspecified value.
+
+\defineentry{parallel:execute}
+\begin{procedure}
+  \code{(parallel:execute \opt{\var{options}} \var{thunks})}
+\end{procedure}
+\returns{} a list of results
+
+The \code{parallel:execute} procedure takes a list of \var{thunks}
+(procedures of no arguments). \code{parallel:execute} returns a list
+of resulting values. Each \var{procedure} may run concurrently unless
+\code{start-limit} is 1.
+
+\defineentry{parallel:execute"!}
+\begin{procedure}
+  \code{(parallel:execute! \opt{\var{options}} \var{thunks})}
+\end{procedure}
+\returns{} unspecified
+
+The \code{parallel:execute!} procedure takes a list of \var{thunks}
+(procedures of no arguments). \code{parallel:execute!} returns an
+unspecified value. Each \var{procedure} may run concurrently unless
+\code{start-limit} is 1.
+
+\defineentry{parallel:for-effect}
+\begin{procedure}
+  \code{(parallel:for-effect \opt{\var{options}} \var{procedure} \var{list$_1$} \var{list$_2$} \etc)}
+\end{procedure}
+\returns{} unspecified
+
+The \code{parallel:for-effect} procedure works like the
+\code{parallel:map} procedure except that it does not accumulate and
+return a list of values. Each application of \var{procedure} may run
+concurrently unless \code{start-limit} is 1.
+
+\defineentry{parallel:map}
+\begin{procedure}
+  \code{(parallel:map \opt{\var{options}} \var{procedure} \var{list$_1$} \var{list$_2$} \etc)}
+\end{procedure}
+\returns{} a list of results
+
+The \code{parallel:map} procedure is similar to Chez Scheme's
```

Comment:
"is similar to" => "works like" for consistency with the other procedures

Reply:
Updated as suggested.